app/Http/Controllers/BookingAPIController.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Booking;
+use App\Models\Room;
+use Carbon\CarbonPeriod;
+use http\Env\Response;
+use Illuminate\Http\Request;
+
+class BookingAPIController extends Controller
+{
+    public function create(Request $request)
+    {
+
+        $request->validate([
+            'room_id' => 'required|integer|exists:rooms,id',
+            'customer' => 'required|string|max:255',
+            'guests' => 'required|integer|min:1',
+            'start' => 'required|date|date_format:Y-m-d|after_or_equal:today',
+            'end' => 'required|date|date_format:Y-m-d|after:start',
```

Comment:
I had no idea these after options existed - that's cool! 

It will mean that your API doesn't return what the front-end is expecting when the dates are the wrong way around, so the front-end probably won't display that error message correctly.

Honestly it's up to you whether or not that's an issue - personally I think this is a really cool solution that I'm very happy to accept.

Reply:
Cool solution stays cool 😎